### PR TITLE
doc: tproxy destination fixes

### PR DESCRIPTION
**Pull request by DanStough**
This PR makes some small tweaks based on internal feedback on the new docs for Transparent Proxy Destinations.

### PR Checklist

* [ ] ~updated test coverage~
* [X] external facing docs updated
* [X] not a security concern

### CODE REVIEW

File: website/content/docs/k8s/connect/terminating-gateways.mdx [hunk 1]
```diff
@@ -261,19 +263,20 @@ spec:
 
 </CodeBlockConfig>
 
--> **NOTE**: If TLS is enabled for external services registered through the Consul catalog, you must include the `caFile` parameter that points to the system trust store of the terminating gateway container.
+-> **NOTE**: If TLS is enabled for external services registered through the Consul catalog and you are not using [transparent proxy `Destinations`](#register-an-external-service-as-a-destination), you must include the `caFile` parameter that points to the system trust store of the terminating gateway container.
 By default, the trust store is located in the `/etc/ssl/certs/ca-certificates.crt` directory.
 Configure the `caFile` parameter to point to the `/etc/ssl/cert.pem` directory if TLS is enabled and you are using one of the following components:
  * Consul Helm chart 0.43 or older
  * Or an Envoy image with an alpine base image
+For Transparent Proxy Destinations, see the [guidance in the table above](#register-an-external-service-as-a-destination).
```
Comment (lkysow) [inline website/content/docs/k8s/connect/terminating-gateways.mdx:271]: CleanShot 2022-08-22 at 12 03 57

The rendering is messed up here. Part of the note has text it shouldn't and the last bullet has text it shouldn't.

File: website/content/docs/k8s/connect/terminating-gateways.mdx [hunk 2]
```diff
@@ -261,19 +263,20 @@ spec:
 
 </CodeBlockConfig>
 
--> **NOTE**: If TLS is enabled for external services registered through the Consul catalog, you must include the `caFile` parameter that points to the system trust store of the terminating gateway container.
+-> **NOTE**: If TLS is enabled for external services registered through the Consul catalog and you are not using [transparent proxy `Destinations`](#register-an-external-service-as-a-destination), you must include the `caFile` parameter that points to the system trust store of the terminating gateway container.
```
Comment (lkysow) [inline website/content/docs/k8s/connect/terminating-gateways.mdx:266]: What `caFile` parameter is this talking about? I know that this was here before your change but maybe we can link to the param? It's confusing reading this what it's talking about.

File: website/content/docs/k8s/connect/terminating-gateways.mdx [hunk 3]
```diff
@@ -291,6 +294,7 @@ spec:
 ```
 
 </CodeBlockConfig>
+-> **NOTE**: L7 Intentions are not currently supported for transparent proxy Destinations.
```
Comment (lkysow) [inline website/content/docs/k8s/connect/terminating-gateways.mdx:297]: What are L7 Intentions? There's not a good page to link to unfortunately but we could at least link to https://consul-17na2ubrq-hashicorp.vercel.app/docs/connect/config-entries/service-intentions#permissions

**Review by lkysow — COMMENTED** [review]
I know these seem nitpicky but I'm worried that right now users won't figure out how to actually use this amazing feature.

* [Here](https://github.com/hashicorp/consul/pull/14281/files#diff-410d7d4b9c3c94aa3b4a160a0032b52d42b34f1237fc16e4615942e014ffe7e0R92) can we say that this is the recommended way if transparent proxy is enabled (the default)? It's not clear for the user which way to choose/which way is best. 
* We use `Destination` everywhere but the actual field in k8s is lowercase: `destination`.
* When saying "if tproxy is enabled" we should link to the tproxy docs page: https://consul-17na2ubrq-hashicorp.vercel.app/docs/connect/transparent-proxy not the config setting because the config setting doesn't tell you enough.
* We shouldn't use `service-defaults` for the k8s docs, the config entry kind is `ServiceDefaults` in k8s.
* Is there a feature name for this? Is it "Destinations", is it "service-defaults destinations", is it "transparent proxy Destinations"? We should stick to one.
*  CleanShot 2022-08-22 at 11 58 37 isn't rendering properly.

**Comment by DanStough** [comment]
@lkysow, I appreciate the thorough review and agree on all points. Let's see if the proposed revisions get us closer to the desired consistency.

File: website/content/docs/k8s/connect/terminating-gateways.mdx [hunk 4]
```diff
@@ -133,10 +135,10 @@ Create a `service-defaults` custom resource for the external service:
 Apply the `ServiceDefaults` resource with `kubectl apply`:
 
 ```shell-session
-$ kubectl apply --filename service-defaults.yaml
+$ kubectl apply --filename serviceDefaults.yaml
```
Comment (lkysow) [inline website/content/docs/k8s/connect/terminating-gateways.mdx:138]: not a problem but the filename is fine to be `-` separated and is pretty normal for k8s yaml files

**Review by lkysow — APPROVED** [review]

**Pull request merged**